lib/zeitwerk/loader.rb

```diff
@@ -262,6 +266,23 @@ def collapse(*glob_patterns)
       end
     end
 
+    # Configure a block to be invoked once a certain constant path is loaded.
+    # Supports multiple callbacks, and if there are many, they are executed in
+    # the order in which they were defined.
+    #
+    #   loader.on_load("SomeApiClient") do
+    #     SomeApiClient.endpoint = "https://my-api.dev"
+    #   end
+    #
+    # @sig (String) { () -> void } -> void
+    def on_load(cpath, &block)
+      raise Error, "on_load only accepts strings" unless cpath.is_a?(String)
```

Comment:
Might make sense to raise a `::TypeError` here.

Reply:
Good point.